docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,183 @@
+-----------------------
+
+The real and imaginary values of a complex tensor can be accessed using the :attr:`real` and
+:attr:`imag`.
+
+.. note::
+     Accessing `real` and `imag` attributes doesn't allocate any memory, and in-place updates on the
+     `real` and `imag` tensors will update the original complex tensor. Also, the
+     returned `real` and `imag` tensors are not contiguous.
+
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+     >>> y.real.mul_(2)
+     tensor([ 1.2250, -0.7546, -0.1722])
+     >>> y
+     tensor([ 1.2250-0.1681j, -0.7546+1.3487j, -0.1722-0.7981j])
+     >>> y.real.stride()
+     (2,)
+
+Angle and abs
+-------------
+
+The angle and absolute values of a complex tensor can be computed using :func:`torch.angle` and
+`torch.abs`.
+
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Linear Algebra
+--------------
+
+Currently, there is very minimal linear algebra operation support for complex tensors.
+We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
+(the latter three are only supported on CPU). However we are working to add support for more
+functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`, :func:`torch.eig`,
+:func:`torch.symeig`. If any of these would help your use case, please
+`search <https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+complex>`_
+if an issue has already been filed and if not, `file one <https://github.com/pytorch/pytorch/issues/new/choose>`_.
+
+
+Serialization
+-------------
+
+Complex tensors can be serialized, allowing data to be saved as complex values.
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+
+Autograd
+--------
+
+PyTorch supports autograd for complex tensors. The autograd APIs can be
+used for both holomorphic and non-holomorphic functions. For non-holomorphic
+functions, the gradient is evaluated assuming the input function is holomorphic.
+For more details, check out the note :ref:`complex_autograd-doc`.
+
+Gradient calculation can also be easily done for functions not supported for complex tensors
+yet by enclosing the unsupported operations between :func:`torch.view_as_real` and
+:func:`torch.view_as_complex` functions. The example shown below computes the dot product
+of two complex tensors, by performing operations on complex tensors viewed as real tensors.
+As shown below, the gradients computed have the same value as you would get if you were to perform
+the operations on complex tensors.
+
+::
+     >>> # computes the complex dot product for complex vectors
+     >>> # represented as float vectors
+     >>> # math: for complex numbers a and b vdot(a, b) = a.conj() * b
```

Comment:
torch.sum(a.conj() * b)) ?

Reply:
synced offline: remove the code block from autograd section until we have a clear definition of functions that can be used with `view_as_real` and `view_as_complex` and for which we can ensure correct gradient computation.